javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -594,59 +599,70 @@ private Vector(int indexShift, HashArrayMappedTrie<Integer, T> trie) {
     @Override
     public Vector<T> distinctBy(Comparator<? super T> comparator) {
         Objects.requireNonNull(comparator, "comparator is null");
-        final java.util.Set<T> seen = new java.util.TreeSet<>(comparator);
+        final Set<T> seen = new java.util.TreeSet<>(comparator);
         return filter(seen::add);
     }
 
     @Override
     public <U> Vector<T> distinctBy(Function<? super T, ? extends U> keyExtractor) {
         Objects.requireNonNull(keyExtractor, "keyExtractor is null");
-        final java.util.Set<U> seen = new java.util.HashSet<>();
+        final Set<U> seen = new java.util.HashSet<>();
```

Comment:
Hint: In Javaslang code I almost always fully qualify Java collection types to make the difference clear, even if there is no conflict.